src/core/openthread.cpp

```diff
@@ -1151,6 +1151,7 @@ uint32_t otGetPollPeriod(otInstance *aInstance)
 
 void otSetPollPeriod(otInstance *aInstance, uint32_t aPollPeriod)
 {
+    aInstance->mThreadNetif.GetMle().SetTimeout(Timer::MsecToSec(aPollPeriod * Mle::kMaxChildKeepAliveAttempts));
```

Comment:
We intentionally allow the poll period independently from the child timeout.  This allows a device to adjust its polling period based on application needs without needing to update the parent with the timeout value.  For example, when a sleepy end device is expecting a response.  If the application truly wants to update the child timeout, then it should call `otSetChildTimeout()`.  Thoughts?